lib/reek/smells/smell_detector.rb

```diff
@@ -60,41 +38,20 @@ def smell_type
         self.class.smell_type
       end
 
-      class << self
-        def smell_category
-          @smell_category ||= default_smell_category
-        end
-
-        def smell_type
-          @smell_type ||= default_smell_category
-        end
-
-        def default_smell_category
-          name.split(/::/)[-1]
-        end
-      end
-
-      def initialize(config = {})
-        config = self.class.default_config.merge(config)
-        @config = SmellConfiguration.new(config)
-        @smells_found = []
-      end
-
       def register(hooks)
         return unless config.enabled?
         self.class.contexts.each { |ctx| hooks[ctx] << self }
       end
 
-      def examine(context)
-        return unless enabled_for? context
+      def run_for(context)
```

Comment:
Changed the naming. I think "examine" was a pretty bad and confusing name since the detectors itself were using "examine_context" and the difference was unclear.